core/story-api/src/main/java/org/alice/interact/manipulator/ObjectRotateDragManipulator.java

```diff
@@ -272,30 +265,28 @@ protected Angle getRotationBasedOnMouse(Point mouseLocation) {
 
   @Override
   public void doDataUpdateManipulator(InputState currentInput, InputState previousInput) {
-    if (!currentInput.getMouseLocation().equals(previousInput.getMouseLocation())) {
-      if (!this.hidCursor) {
-        this.hideCursor();
-      }
-      Angle currentAngle = getRotationBasedOnMouse(currentInput.getMouseLocation());
-      if ((currentAngle != null) && (this.originalAngleBasedOnMouse != null)) {
-        Angle angleDif = AngleUtilities.createSubtraction(currentAngle, this.originalAngleBasedOnMouse);
-        //The angleDif is the amount the object as rotated relative to the start of the manipulation
-        //By snapping on angleDif, we're snapping to snap angles relative to the orientation at the start of the manipulation
-        Angle snappedAngle = SnapUtilities.doRotationSnapping(angleDif, this.dragAdapter);
-        boolean didSnap = snappedAngle.getAsDegrees() != angleDif.getAsDegrees();
-        if (didSnap) {
-          angleDif = snappedAngle;
-        }
-
-        this.manipulatedTransformable.setLocalTransformation(this.originalLocalTransformation);
-        this.manipulatedTransformable.applyRotationAboutArbitraryAxis(this.rotationHandle.getRotationAxis(), angleDif, this.rotationHandle.getReferenceFrame());
-        manipulatedTransformable.notifyTransformationListeners();
-
-        if (didSnap) {
-          SnapUtilities.showSnapRotation(this.rotationHandle);
-        } else {
-          SnapUtilities.hideRotationSnapVisualization();
-        }
+    if (currentInput.getMouseLocation().equals(previousInput.getMouseLocation())) {
+      return;
+    }
+    if (!this.hidCursor) {
+      this.hideCursor();
+    }
+    Angle currentAngle = getRotationBasedOnMouse(currentInput.getMouseLocation());
+    if ((currentAngle != null) && (this.originalAngleBasedOnMouse != null)) {
+      Angle angleDif = currentAngle.minus(this.originalAngleBasedOnMouse);
+      //The angleDif is the amount the object as rotated relative to the start of the manipulation
+      //By snapping on angleDif, we're snapping to snap angles relative to the orientation at the start of the manipulation
+      Angle snappedAngle = SnapUtilities.doRotationSnapping(angleDif, this.dragAdapter);
+      boolean didSnap = !snappedAngle.isCloseTo(angleDif);
+
+      this.manipulatedTransformable.setLocalTransformation(this.originalLocalTransformation);
+      this.manipulatedTransformable.applyRotationAboutArbitraryAxis(this.rotationHandle.getRotationAxis(), snappedAngle, this.rotationHandle.getReferenceFrame());
+      manipulatedTransformable.notifyTransformationListeners();
+
+      if (didSnap) {
```

Comment:
we don't actually use didSnap til this check, could just move the whole isCloseTo into this if statement?

Reply:
Done. Added to branch.